File: 简化v1_26.5.7/error-reviewer.js
const REVIEW_SYSTEM_PROMPT = `
你是一位极其严谨的零售数据审计专家。你的任务是“检查初级报告中是否有明显数据异常”，进行二次复核。
你需要重点关注以下几类常见分析错误，并根据提供给你的【底层原始数据】进行核对：
1. **周期换算错误**：比如年/月下滑 20% 是否仅仅是因为当前还没到年/月底？（例如今天是9月2日，9月的当前累积数据如果直接和8月全月比，必然是下滑的）。如果是这种情况，请按天数比例换算过来（等效完整年/月收入），看看等效后的数据是否有参考性，具体数值大概是多少。
2. **无意义商品的干扰**：检查初级报告里提到的“热销商品”或“缺货商品”，是否有矿泉水、普通塑料袋等对整体经营诊断无重大指导意义的低毛利/高频凑单商品影响了判断？如果有，请指出来。
3. **数据自相矛盾**：初级报告中的结论是否与提供的数据明显冲突？（比如数据明明是上涨的，报告却说是下跌；或者提到某个商品缺货，但底层数据里明确显示有货）。
4. **因缺少数据导致的无法分析或分析不到位**：比如用户导入数据过少，导致某类型数据分析后异常波动或表现，不应作为参考。
5. **推荐行动的可执行性**：要考虑到当前店定位，再考虑到环境和常识，避免一些不切实际的建议

**输出要求：**
- 你的输出应当是纯文本形式的【评审意见】。
- 如果初级报告没问题，只需回答：“错误审核通过，暂未发现明显逻辑或计算谬误。”
- 如果发现了问题，请逐条列出：“发现异常：1. ... 2. ...”，明确指出初级报告哪里说错了，并在后续的详细重写阶段建议 AI 纠正。
- 请直接输出结论，无需包含过多的客套话。
`;

async function reviewError(settings, report, cleanedDataTexts, onChunk) {
    if (!settings || !settings.apiKey) {
        return "初级分析报告 - 错误审核：未配置 API Key，已跳过真实审核，采用模拟通过。";
    }

    const { baseUrl, apiKey, model } = settings;

    // 构建基础上下文信息
    const now = new Date();
    const dateStr = now.toLocaleDateString('zh-CN', { year: 'numeric', month: 'long', day: 'numeric', weekday: 'long' });
    const timeStr = now.toLocaleTimeString('zh-CN', { hour: '2-digit', minute: '2-digit' });

    const contextHeader = `【当前分析环境】
- 城市：福州
- 日期：${dateStr}
- 时间：${timeStr}
`;

    const userContent = contextHeader + 
        '\n\n【初级分析报告】\n' + report + 
        '\n\n【底层原始数据】\n' + (cleanedDataTexts ? cleanedDataTexts.join('\n\n---\n\n') : '暂无底层数据');

    const url = baseUrl.replace(/\/+$/, '') + '/chat/completions';
    const payload = {
        model,
        messages: [
            { role: 'system', content: REVIEW_SYSTEM_PROMPT },
            { role: 'user', content: userContent }
        ],
        temperature: 0.2,
        reasoning_effort: "medium",
        max_tokens: 8192,
        stream: true,
        stream_options: { include_usage: true }
    };

    const payloadSize = JSON.stringify(payload).length;
    if (onChunk) onChunk(`[审计节点] 准备发送流式请求，Payload大小: ${Math.round(payloadSize / 1024)} KB\n`);

    const controller = new AbortController();
    const timeoutId = setTimeout(() => controller.abort(), 120000);
    
    if (settings && settings.signal) {
        settings.signal.addEventListener('abort', () => controller.abort());
    }

    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${apiKey}`
            },
            body: JSON.stringify(payload),
            signal: controller.signal
        });

        clearTimeout(timeoutId);

        if (!response.ok) {
            const errText = await response.text();
            throw new Error(`HTTP ${response.status}: ${errText}`);
        }

        const reader = response.body.getReader();
        const decoder = new TextDecoder('utf-8');
        let fullText = '';
        let buffer = '';
        let finalJson = null;

        while (true) {
            const { done, value } = await reader.read();
            if (done) break;
            
            buffer += decoder.decode(value, { stream: true });
            let lines = buffer.split('\n');
            buffer = lines.pop(); // 保持最后一行不完整的在 buffer 中

            for (let line of lines) {
                line = line.trim();
                if (line.startsWith('data:')) {
                    const dataStr = line.substring(5).trim();
                    if (dataStr === '[DONE]') continue;
                    try {
                        const dataObj = JSON.parse(dataStr);
                        if (dataObj.choices && dataObj.choices[0].delta && dataObj.choices[0].delta.content) {
                            fullText += dataObj.choices[0].delta.content;
                        }
                        if (dataObj.usage) finalJson = dataObj;
                        else if (!finalJson) finalJson = dataObj;
                    } catch (e) {
                        // 忽略不完整的 JSON 解析错误
                    }
                }
            }
        }

        // 构造一个兼容旧逻辑的返回结构
        if (finalJson) {
            finalJson.choices = [{ message: { content: fullText } }];
            return finalJson;
        }
        return fullText;
    } catch (err) {
        clearTimeout(timeoutId);
        let errMsg = err.message;
        if (err.name === 'AbortError') errMsg = '请求超时 (超过120秒)，推理模型思考时间过长';
        
        console.error("错误评审 AI 流式调用失败:", err);
        return `错误审核失败。原因：${errMsg} | Payload: ${Math.round(payloadSize / 1024)}KB`;
    }
}

module.exports = { reviewError };

File: packages/ai/error_reviewer.py
"""
error_reviewer.py — 错误评审模块 (v2: 场景感知 + 格式化输入)
"""
import json
from datetime import datetime
from .ai_caller import shared_stream_fetch


def _build_review_prompt(scene: dict = None) -> str:
    return """你是一位极其严谨的数据审计专家。任务：复核初级报告中是否存在明显错误。

重点检查：
1. 数据引用错误：报告引用的数值是否与【证据包】中的数据一致？
2. 周期换算错误：环比/月变化结论是否考虑了时间口径（如未到月底）？
3. 指标交叉矛盾：不同指标的结论是否互相矛盾（如营收涨但毛利跌未解释）？
4. 缺数据误判：标记为 uncountable 的指标是否在报告中被引用或编造？
5. 建议可行性：行动方案是否符合当前数据能力和行业常识？
6. 干扰信息过滤：排名、排序、ID等元数据字段不应作为经营判断依据

输出要求：
- 纯文本评审意见
- 没问题 → "审核通过，未发现明显逻辑或数据错误。"
- 有问题 → 逐条列出 "发现异常：1. ... 2. ..."，每条注明引用哪条证据
- 直接输出结论，无客套话"""


def _format_evidence_for_review(evidence_bundle: dict) -> str:
    items = evidence_bundle.get("items", [])
    if not items:
        return "暂无证据数据。"

    lines = ["【证据包（复核基准）】"]
    lines.append(f"共 {len(items)} 条证据:")
    for item in items:
        metric_id = item.get("metric_id", "?")
        title = item.get("title", "?")
        status = item.get("status", "?")
        val = item.get("value")
        if isinstance(val, (dict, list)):
            val_str = json.dumps(val, ensure_ascii=False)[:120]
        else:
            val_str = str(val)[:80] if val is not None else "-"
        lines.append(f"  [{status}] {title} | value={val_str}")
    return "\n".join(lines)


async def review_error(settings: dict, report: str, cleaned_data_texts: list = None) -> dict | str:
    """错误评审 (兼容旧管线)"""
    if not settings or not settings.get("apiKey"):
        return "审核通过：未配置 API Key，已跳过。"

    now = datetime.now()
    context = f"【分析环境】\n- 日期：{now.strftime('%Y年%m月%d日')} {now.strftime('%H:%M')}\n"
    raw_str = "\n\n---\n\n".join(cleaned_data_texts) if cleaned_data_texts else "暂无底层数据"
    user_content = context + "\n\n【初级分析报告】\n" + (report or "") + "\n\n【底层数据】\n" + raw_str

    reasoning_effort = "medium"
    if isinstance(settings, dict):
        raw = settings.get("reasoningEffort") or settings.get("reasoning_effort") or ""
        r = str(raw).strip().lower()
        if r in {"low", "medium", "high"}:
            reasoning_effort = r

    try:
        return await shared_stream_fetch(settings["baseUrl"], settings["apiKey"], {
            "model": settings["model"],
            "messages": [
                {"role": "system", "content": _build_review_prompt()},
                {"role": "user", "content": user_content},
            ],
            "temperature": 0.2, "reasoning_effort": reasoning_effort, "max_tokens": 8192,
        })
    except Exception as e:
        return f"错误审核失败。原因：{str(e)}"


async def review_error_new(settings: dict, scene: dict, report: str, evidence: dict) -> dict | str:
    """错误评审 (新管线 — 格式化输入)"""
    if not settings or not settings.get("apiKey"):
        return "审核通过：未配置 API Key，已跳过。"

    reasoning_effort = "medium"
    if isinstance(settings, dict):
        raw = settings.get("reasoningEffort") or settings.get("reasoning_effort") or ""
        r = str(raw).strip().lower()
        if r in {"low", "medium", "high"}:
            reasoning_effort = r

    now = datetime.now()
    parts = [
        f"【分析环境】\n- 行业：{(scene or {}).get('industry', 'unknown')}\n- 日期：{now.strftime('%Y年%m月%d日 %H:%M')}",
        "",
        f"【初级分析报告】\n{report}",
        "",
        _format_evidence_for_review(evidence),
    ]
    user_content = "\n".join(parts)

    try:
        return await shared_stream_fetch(settings["baseUrl"], settings["apiKey"], {
            "model": settings["model"],
            "messages": [
                {"role": "system", "content": _build_review_prompt(scene)},
                {"role": "user", "content": user_content},
            ],
            "temperature": 0.2, "reasoning_effort": reasoning_effort, "max_tokens": 8192,
        })
    except Exception as e:
        return f"错误审核失败。原因：{str(e)}"
